src/s3ql/backends/b2/b2_backend.py

```diff
@@ -0,0 +1,807 @@
+import base64
+from dugong import (HTTPConnection, CaseInsensitiveDict, is_temp_network_error, BodyFollowing,
+                    ConnectionClosed, ConnectionTimedOut)
+from urllib.parse import urlparse
+import urllib
+import json
+import re
+import ssl
+import binascii
+from ast import literal_eval
+from itertools import count
+import io
+import os
+from shutil import copyfileobj
+
+from  typing import Dict, Tuple, Union, List
```

Comment:
Is it possible to start using PEP 484, since the minimum python version is 3.6 now?

Reply:
> Is it possible to start using PEP 484, since the minimum python version is 3.6 now?

@Nikratio what do you think about using [PEP 484](https://www.python.org/dev/peps/pep-0484/)? Incremental (new/changed code only)? Refactor everything in one go? Not using type hints at all?

Reply:
I think adding type hints would be great.